cmd/amp2response.cpp

```diff
@@ -297,6 +297,8 @@ void run ()
 
         // Generate the ZSH -> amplitude transform
         Eigen::MatrixXd transform = Math::ZSH::init_amp_transform<default_type> (rotated_dirs_azel.col(1), lmax[shell_index]);
+        if (!transform.allFinite())
+          throw Exception ("Unable to construct A2SH transformation for shell b=" + str(int(std::round((*shells)[shell_index].get_mean()))) + "; lmax (" + str(lmax[shell_index]) + ") may be too large for this shell");
```

Comment:
You could break that up over multiple lines if you want, it might be cleaner:
```C++
if (!transform.allFinite()) {
  Exception excp ("Unable to construct A2SH transformation for shell b=" + str(int(std::round((*shells)[shell_index].get_mean()))));
  excp.push_back ("  lmax (" + str(lmax[shell_index]) + ") may be too large for this shell");
  throw excp;
}
```